hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -615,9 +615,13 @@ object HoodieFileIndex extends Logging {
   }
 
   // if database name is not set, fall back to use 'default' instead of failing
-  def getDatabaseName(tableConfig: HoodieTableConfig)  = {
+  def getDatabaseName(tableConfig: HoodieTableConfig, defaultDatabase: String)  = {
```

Comment:
in line 541 can we just make sure the default data base is never null, and in this method always use the passed default database as a fallback.

Reply:
I think this is minor and we can keep the deciding logic within the method so it's easier to understand.